### ISSUE TITLE
CPU memory keep going up

### ISSUE THREAD

**Issue by HanXuMartin** [open-mmlab/mmdetection#10818]
Hi, I use RTMDet on my own dataset and the cpu memory keep going up during training and the process finally get killed. My dataset contains 700K 1920*1080 images  and is formatted into CocoDataset. I tried to use less workers but it doesn't work.
Here is my dataloader config
train_dataloader = dict(
    batch_size=128,
    num_workers=2,
    pin_memory=False,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    batch_sampler=None,
    dataset=dict(
        type=dataset_type,
        data_root = data_root,
        ann_file = f"{anno_file}/train.json",
        data_prefix=dict(img=''),
        metainfo=metainfo,
        serialize_data=False,
        filter_cfg=dict(filter_empty_gt=True, min_size=5),
        pipeline=[

Any suggestions?

**Comment by hhaAndroid** [comment]
@HanXuMartin Such a high resolution input indeed requires a significant amount of memory and GPU memory. You should consider reducing the batch_size.

**Comment by HanXuMartin** [comment]
@hhaAndroid Thanks for your reply. I decreased the batch size to 32 and use MemoryProfilerHook to monitor the memory usage. There are still some questions:
1. the memory utilization keep increasing during the training.
2. the memory utilization does not decrease when one epoch ends. 
I don't think this is correct. Is there some cpu memory leaking? or just because my cpu can not support this amount of data?

**Comment by hhaAndroid** [comment]
@HanXuMartin This is indeed unreasonable. We can check if there are any other communities that have encountered this issue in the future.

**Comment by Chop1** [comment]
I have the same issue. CPU memory keep going up until CPU OOM

**Comment by gongjizhang** [comment]
Same issue. CPU memory keep going up when I fine tune grounding dino！
